### PR TITLE
Add migration to special field

### LINKED ISSUE #6840: Offer migration from special fields in keywords to special fields as fields
Refs https://github.com/JabRef/jabref/issues/6621

### PR DESCRIPTION
**Pull request by DominikVoigt**
This PR migrates all occurrences of special field values in the keywords into their respective special fields.
Fixes #6840.

- [x] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by tobiasdiez — DISMISSED** [review]
Thanks! Looks good to me, without any of the usual comments and suggestions ;-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 ### Changed
 
 - The content of the field `timestamp` is migrated to `creationdate`. In case one configured "udpate timestampe", it is migrated to `modificationdate`. [koppor#130](https://github.com/koppor/jabref/issues/130)
+- The JabRef specific meta-data content in the main field such as priorities (prio1, prio2, ...) are migrated to their respective fields. They are removed from the keywords. [#6840]((https://github.com/jabref/jabref/issues/6840))
```
Comment (Siedlerchr) [inline CHANGELOG.md:23]:
```suggestion
- The JabRef specific meta-data content in the main field such as priorities (prio1, prio2, ...) are migrated to their respective fields. They are removed from the keywords. [#6840](https://github.com/jabref/jabref/issues/6840)
```

**Review by Siedlerchr — DISMISSED** [review]
lgtm as well

File: src/test/java/org/jabref/migrations/SpecialFieldsToSeparateFieldsTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Test
+    public void migrateMultipleSpecialFields() {
+        BibEntry entry = new BibEntry().withField(StandardField.AUTHOR, "JabRef")
+                                       .withField(StandardField.KEYWORDS, "printed, prio1");
+        new SpecialFieldsToSeparateFields(',').performMigration(new ParserResult(List.of(entry)));
+
+        assertEquals("prio1", entry.getField(SpecialField.PRIORITY).get());
+        assertEquals("printed", entry.getField(SpecialField.PRINTED).get());
+
```
Comment (koppor) [inline src/test/java/org/jabref/migrations/SpecialFieldsToSeparateFieldsTest.java:49]: It should be checked, whether the keyword field has `printed` and `prio1` removed.

Why not checking an expected BibEntry? --> Applies for all test cases 😇

**Review by koppor — CHANGES_REQUESTED** [review]
I think, this is a very short implementation. Looks good.

Could you remove org.jabref.logic.specialfields.SpecialFieldsUtils and org.jabref.logic.specialfields.SpecialFieldsUtilsTest, because this migration should make them obsolete.

File: src/main/resources/build.properties [hunk 3]
```diff
@@ -2,6 +2,6 @@ version=${version}
 year=${year}
 maintainers=${maintainers}
 azureInstrumentationKey=${azureInstrumentationKey}
-springerNatureAPIKey=${springerNatureAPIKey}
-astrophysicsDataSystemAPIKey=${astrophysicsDataSystemAPIKey}
-ieeeAPIKey=${ieeeAPIKey}
+springerNatureAPIKey=77618eaf1abca4f60cd365a717bb68c7
```
Comment (tobiasdiez) [inline src/main/resources/build.properties:5]: Please remove this (and hope that no one else saw them ;-) )

**Review by koppor — APPROVED** [review]
I was thinking, if we should check for `importFormatPreferences.isKeywordSyncEnabled()`.

The "risk" is that "common" words such as `read`, `skimmed`, `revelant`, and `printed` could always automatically be moved from keywords to special fields and users will complain that keywords are lost. I think, this risk is low, thus I approve.

**Pull request merged**